src/Core/Geographic/Extent.js

```diff
@@ -354,11 +354,13 @@ Extent.prototype.intersect = function intersect(other) {
     if (!this.intersectsExtent(other)) {
         return new Extent(this.crs(), 0, 0, 0, 0);
     }
-    return new Extent(this.crs(),
+    const extent = new Extent(this.crs(),
         Math.max(this.west(), other.west(this._internalStorageUnit)),
         Math.min(this.east(), other.east(this._internalStorageUnit)),
         Math.max(this.south(), other.south(this._internalStorageUnit)),
         Math.min(this.north(), other.north(this._internalStorageUnit)));
+    extent._internalStorageUnit = this._internalStorageUnit;
```

Comment:
As always, we should avoid accessing a private member like this. As I saw that this is used heavily in other files, what about adding an accessor to it or making it public ?

Reply:
Btw, in the team, we have different interpretation of this private `_`, some of us think of it as a project-wide private. I personnally prefer the strict interpretation of class/module-wide private à la Java. We could also find a middle-ground and use it like it was a java `protected` field (so package-private), but that would mean having a directory structure that would look like packages. In my opinion, it's not the case yet in iTowns, and I'm not sure it's worth the trouble.

Reply:
Thanks, good to know. However note that `_internalStorageUnit` is used in `OBB` and `Coordinates`, among other things.

Reply:
yep, that is subject to debate indeed. 

For `OBB`, this should be an additional optional constructor parameter I  guess. 

For coordinates, I'm less disturbed by it, as `Coordinates` and `Extent` are very much linked together in their logic (and they are in the same `geographic` "package"). A solution would be to have accessors of course, or maybe move `offsetInExtent` in `Extent.js` ? Or even make them a single module... Or leave it like that, imo it's not a big problem, no strong opinion there :-)